org/postgresql/geometric/PGline.java

```diff
@@ -17,38 +18,67 @@
 import java.sql.SQLException;
 
 /**
- * This implements a line consisting of two points.
+ * This implements a line represented by the linear equation Ax + By + C = 0
  *
- * Currently line is not yet implemented in the backend, but this class
- * ensures that when it's done were ready for it.
- */
+ **/
 public class PGline extends PGobject implements Serializable, Cloneable
 {
+
+    /**
+     * Coefficient of x
+     */
+    public double a;
+
+    /**
+     * Coefficient of y
+     */
+    public double b;
+
     /**
-     * These are the two points.
+     * Constant
      */
-    public PGpoint point[] = new PGpoint[2];
+    public double c;
 
     /**
-     * @param x1 coordinate for first point
-     * @param y1 coordinate for first point
-     * @param x2 coordinate for second point
-     * @param y2 coordinate for second point
+     * @param a coefficient of x
+     * @param b coefficient of y
+     * @param c constant
+     */
+    public PGline(double a, double b, double c) {
+        this();
+        this.a = a;
+        this.b = b;
+        this.c = c;
+    }
+
+    /**
+     * @param x1 coordinate for first point on the line
+     * @param y1 coordinate for first point on the line
+     * @param x2 coordinate for second point on the line
+     * @param y2 coordinate for second point on the line
      */
     public PGline(double x1, double y1, double x2, double y2)
     {
-        this(new PGpoint(x1, y1), new PGpoint(x2, y2));
+        this();
+        double a = (y2 - y1) / (x2 - x1);
```

Comment:
Hi Phillip,

Any chance I could get you to add a test case for this as well ?

Dave

Dave Cramer

On 29 June 2015 at 21:44, Phillip Ross <notifications@github.com> wrote:

> In org/postgresql/geometric/PGline.java
> <https://github.com/pgjdbc/pgjdbc/pull/343#discussion_r33533728>:
>
> >       */
> >      public PGline(double x1, double y1, double x2, double y2)
> >      {
> > -        this(new PGpoint(x1, y1), new PGpoint(x2, y2));
> > +        this();
> > +        double a = (y2 - y1) / (x2 - x1);
>
> After a little investigation... it appears postgresql doesn't mind being
> given NaN, inf, or -inf so there's really no handling that needs to be done
> otherwise. Since these are double precision values, there will be no
> exceptions thrown. Regardless, the constructor does need changed to handle
> vertical lines. I'll push another commit for the vertical line handling.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/343/files#r33533728>.
>